corehq/apps/hqwebapp/templatetags/hq_shared_tags.py

```diff
@@ -600,6 +605,17 @@ def html_attr(value):
     return conditional_escape(value)
 
 
+@register.filter
+def language_name_local(lang_code):
+    """
+    Override built-in language_name_local filter to work with languages defined
+    in langcodes and always title case its output.
+    """
+    lang = langs_by_code.get(lang_code)
+    lang_code = lang['two'] if lang else ''
+    return i18n.language_name_local(lang_code).title()
```

Comment:
It seems if a language only has three letter code, this will raise `KeyError: 'Unknown language code .'` Is this handled?

Reply:
This is a very good point, thank you. This shouldn't really happen in practice, but it is possible (and some languages in langcodes don't have two-letter versions, so would be calling `language_name_local` on an empty string that way). [0b87ac8](https://github.com/dimagi/commcare-hq/pull/36733/commits/0b87ac81e162bbb338de908965593e4e259e9516)